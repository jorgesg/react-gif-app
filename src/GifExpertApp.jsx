import {useState} from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['NBA']);

    const onAddCategory = (newCategory) => {

        if(categories.includes(newCategory)) return;

        setCategories([newCategory, ...categories]);
        //setCategories( cat => [...cat, 'Valorant']);
    }

    return (
        <>
            {/* Titulo */}
            <h1>GifExpertApp</h1>

            {/* Input */}
            <AddCategory 
                // setCategories={setCategories}
                onNewCategory={onAddCategory}
            />

            { categories.map(category => (
                    <GifGrid 
                        key={category} 
                        category={category}
                    />
                )
            ) }

        </>
    )
}
