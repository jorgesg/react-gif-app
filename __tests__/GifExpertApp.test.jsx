import { fireEvent, render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Testing <GifExpertApp />', () => {

    test('should match snapshot', () => {
        const { container } = render(<GifExpertApp />);
        
        expect(container).toMatchSnapshot();
    });

    test('should not add categories with length <= 1', () => {
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        fireEvent.input(input, {target: {value: 'a'}});
        expect(screen.getAllByRole('heading', {level: 3}).length).toBe(1);
    });

    test('should add new category', () => {
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        fireEvent.input(input, {target: {value: 'Deftones'}});
        fireEvent.input(input, {target: {value: 'Opeth'}});
        expect(screen.getAllByRole('heading', {level: 3}).length).toBe(1);//3
    });

    test('should NOT add a previously added category', () => {
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        fireEvent.input(input, {target: {value: 'Alien'}});
        fireEvent.input(input, {target: {value: 'Alien'}});
        expect(screen.getAllByRole('heading', {level: 3}).length).toBe(1);//2
    });

})