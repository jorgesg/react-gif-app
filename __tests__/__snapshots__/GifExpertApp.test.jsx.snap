// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp /> should match snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    NBA
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
